### DISCUSSION TITLE
Compiling for Windows 10

### DISCUSSION THREAD

**Discussion by robert-dzikowski** [bitcoin-core/secp256k1#1502] [Q&A] [answered]
I  cross compiled on Ubuntu using
`cmake .. -DCMAKE_TOOLCHAIN_FILE=../cmake/x86_64-w64-mingw32.toolchain.cmake
`

and compiled on Windows 10 using
```
cmake -G "Visual Studio 17 2022" -A x64 -S . -B build
cmake --build build --config RelWithDebInfo
```
In both cases compiled secp256k1 library causes the following error when I try to run Electrum wallet (https://github.com/spesmilo/electrum) on Windows 10
```
File "D:\Data\repos\Python\electrum\electrum\ecc_fast.py", line 151, in <module>
    raise ImportError("Failed to load libsecp256k1")
ImportError: Failed to load libsecp256k1
  0.84 | E | ecc_fast | failed to load libsecp256k1: LibModuleMissing('libsecp256k1 library found but it was built without required module (--enable-module-recovery)')
```
I don't get this error when I compile this library on Ubuntu 22.04.2 LTS using Electrum build script, i.e. Electrum wallet runs without problems on Ubuntu using that compiled library.

**Comment by real-or-random** [comment, marked as answer]
You need to enable the recovery module, which is disabled by default.

Add `-DSECP256K1_ENABLE_MODULE_RECOVERY=ON` to the first `cmake` invocation.

**Comment by robert-dzikowski** [comment]
I wrote
```
cmake -G "Visual Studio 17 2022" -A x64 -S . -B build -DSECP256K1_ENABLE_MODULE_RECOVERY=ON
```
and it worked. Thanks.
